### ISSUE TITLE
RMI tests without destroyed properly and hang on machines

### ISSUE THREAD

**Issue by sophia-guo** [AdoptOpenJDK/openjdk-tests#819]
Some tests triggered rmid and then cannot destroyed it in 600m. So test failed but no other step to force kill the process. When tests passing the issue is hidden.

We have reported this issue and asked for help to report it to openjdk and believe lost tracking.

**Comment by sophia-guo** [comment]
https://github.com/AdoptOpenJDK/openjdk-tests/issues/487

**Comment by sophia-guo** [comment]
https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk8_hs_openjdktest_x86-64_macos/226/

```
06:58:44 TEST FAILED: rmid not destroyed in: 60000 milliseconds
06:58:44 Exception in thread "main" TestFailedException: TEST FAILED: rmid not destroyed in: 60000 milliseconds
06:58:44 	at TestLibrary.bomb(TestLibrary.java:121)
06:58:44 	at TestLibrary.bomb(TestLibrary.java:124)
06:58:44 	at ActivationLibrary.rmidCleanup(ActivationLibrary.java:176)
06:58:44 	at ExtLoadedImplTest.main(ExtLoadedImplTest.java:72)
06:58:44 rerun:
06:58:44 cd /Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/jvmtest/openjdk_regression/work/java/rmi/activation/Activatable/extLoadedImpl/ext && \

```

```
DXU773:~ jenkins$ ps -f -u jenkins
  501 88218     1   0  9:57pm ??         0:55.11 /Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/openjdkbinary/j2sdk-image/jre/bin/java -Djava.security.policy=/Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/openjdk-tests/openjdk_regression/openjdk-jdk/jdk/test/java/rmi/activation/Activatable/extLoadedImpl/rmid.security.policy -Dsun.rmi.server.activation.debugExec=true -Dtest.src=/Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/openjdk-tests/openjdk_regression/openjdk-jdk/jdk/test/java/rmi/activation/Activatable/extLoadedImpl -Dtest.classes=/Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/jvmtest/openjdk_regression/work/classes/1/java/rmi/activation/Activatable/extLoadedImpl -Djava.rmi.server.logLevel=v -Dsun.rmi.activation.execTimeout=60000 sun.rmi.server.Activation -log /Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/jvmtest/openjdk_regression/work/java/rmi/activation/Activatable/extLoadedImpl/ext/./log -port 57513 -C-Dtest.src=/Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/openjdk-tests/openjdk_regression/openjdk-jdk/jdk/test/java/rmi/activation/Activatable/extLoadedImpl -C-Dtest.classes=/Users/jenkins/workspace/openjdk8_hs_openjdktest_x86-64_macos/jvmtest/openjdk_regression/work/classes/1/java/rmi/activation/Activatable/extLoadedImpl -C-Djava.rmi.server.useCodebaseOnly=false
 ```

**Comment by smlambert** [comment]
@karianna - we believe this should get reported upstream, and need help to (from someone who can raise defects in JBS) can you advise?

**Comment by karianna** [comment]
> @karianna - we believe this should get reported upstream, and need help to (from someone who can raise defects in JBS) can you advise?

Yes, we have a JBS Reporting Team!  I'll post it in the slack.

**Comment by karianna** [comment]
> > @karianna - we believe this should get reported upstream, and need help to (from someone who can raise defects in JBS) can you advise?
> 
> Yes, we have a JBS Reporting Team! I'll post it in the slack.

#jbs channel is actually public - so feel free to post links there.

**Comment by stooart-mon** [comment]
I think we'll need more information before reporting it upstream. 
Would it be possible to run "jstack -l <pid>" with <pid> being the pid of the rmid process?
Is there any other information that will narrow this down? What machine/OS/JVM versions does this appear?

**Comment by sophia-guo** [comment]
Hi @stooart-mon, there are two issues here. One is the failed test itself, which is machine/OS/JVM related. The other one is actually machine/OS/JVM independent and it is the one we want to focus on. That is 'the whole set of rmi regression tests clean up'. Tests triggered rmid and then check if it can be destroyed it in 600m. If yes, test pass and no issue. If not test failed and then no any other step to force kill the process or no other step to do the clean up, which leave the process hang. 
The output for "jstack -l" hope it's helpful.

```
DXU773:~ jenkins$ jstack -l 88218
2019-01-24 07:41:26
Full thread dump OpenJDK 64-Bit Server VM (25.202-b08 mixed mode):

"Attach Listener" #14 daemon prio=9 os_prio=31 tid=0x00007f8cc2813000 nid=0x465b runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"GC Daemon" #12 daemon prio=2 os_prio=31 tid=0x00007f8cc2149800 nid=0x4e03 in Object.wait() [0x000000011ee83000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007a00015e0> (a sun.misc.GC$LatencyLock)
	at sun.misc.GC$Daemon.run(GC.java:117)
	- locked <0x00000007a00015e0> (a sun.misc.GC$LatencyLock)

   Locked ownable synchronizers:
	- None

"RMI Reaper" #11 prio=5 os_prio=31 tid=0x00007f8cc2a36800 nid=0x4c03 in Object.wait() [0x000000011ed80000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007a0007690> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	- locked <0x00000007a0007690> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at sun.rmi.transport.ObjectTable$Reaper.run(ObjectTable.java:351)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"RMI TCP Accept-57513" #10 daemon prio=5 os_prio=31 tid=0x00007f8cc2a33000 nid=0x4a07 runnable [0x000000011ec7d000]
   java.lang.Thread.State: RUNNABLE
	at java.net.PlainSocketImpl.socketAccept(Native Method)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(TCPTransport.java:405)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(TCPTransport.java:377)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"Service Thread" #7 daemon prio=9 os_prio=31 tid=0x00007f8cc281a000 nid=0x4103 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C1 CompilerThread1" #6 daemon prio=9 os_prio=31 tid=0x00007f8cc20ca800 nid=0x3f03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread0" #5 daemon prio=9 os_prio=31 tid=0x00007f8cc2089800 nid=0x3d03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 tid=0x00007f8cc2041800 nid=0x2d13 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007f8cc2811800 nid=0x2903 in Object.wait() [0x000000011cd34000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007a0007e30> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	- locked <0x00000007a0007e30> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)

   Locked ownable synchronizers:
	- None

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007f8cc2811000 nid=0x2703 in Object.wait() [0x000000011cc31000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007a0007fe8> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x00000007a0007fe8> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

   Locked ownable synchronizers:
	- None

"main" #1 prio=5 os_prio=31 tid=0x00007f8cc2014800 nid=0x1303 waiting on condition [0x000000010ad00000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at sun.rmi.server.Activation.main(Activation.java:2126)

   Locked ownable synchronizers:
	- None

"VM Thread" os_prio=31 tid=0x00007f8cc2012000 nid=0x2503 runnable 

"GC task thread#0 (ParallelGC)" os_prio=31 tid=0x00007f8cc2803800 nid=0x2103 runnable 

"GC task thread#1 (ParallelGC)" os_prio=31 tid=0x00007f8cc2804000 nid=0x2303 runnable 

"VM Periodic Task Thread" os_prio=31 tid=0x00007f8cc2819000 nid=0x4303 waiting on condition 

JNI global references: 203


```

**Comment by stooart-mon** [comment]
Ok, do you know how you'd fix the issue whereupon the rmid processes are cleanedup?
My own automation does this in it's own particular way, but it only suits my assumptions.

**Comment by smlambert** [comment]
Useful guidance for gathering info for JBS: https://github.com/AdoptOpenJDK/openjdk-tests/wiki/Guidance-for-Creating-OpenJDK-Test-Defects

**Comment by smlambert** [comment]
@sophia-guo is this a dup of #497

**Comment by smlambert** [comment]
This one has been sitting around a while.  If we want to address this we should either suggest a way to update upstream test material, or alternatively add a step in TKG teardown that looks for and kills any dangling processes (and/or perhaps process check in setup).

**Comment by smlambert** [comment]
Closing this in favour of https://github.com/AdoptOpenJDK/TKG/issues/45
